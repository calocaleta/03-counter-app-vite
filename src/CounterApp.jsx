import {useState} from 'react';
import React from 'react';
import PropTypes from 'prop-types'

const CounterApp = ({value}) => {

    console.log('render');

    const [ counter, setCounter] = useState(value);  

    const handleAdd = (event) => {
        //console.log(event);
        //value = 1000;
        //setCounter(counter + 1);
        setCounter((prevValue) => prevValue + 1);
    }

    const handleSubstract = () => setCounter(counter - 1);

    const handleReset = () => {
        setCounter(value);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2 data-testid="test-value"> { counter } </h2>
            <button onClick={ handleAdd }> +1 </button>
            <button onClick={ handleSubstract }> -1 </button>
            <button onClick={ handleReset }> Reset </button>
        </>
    );
};

export default CounterApp;

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}