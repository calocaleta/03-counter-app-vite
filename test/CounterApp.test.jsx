/** @jest-environment jsdom */
import { fireEvent, render,screen } from "@testing-library/react";
import CounterApp from "../src/CounterApp";

describe('Pruebas con el componente CounterApp', () => { 
    const initialValue = 1;

    test('Prueba hacer match con snapshot',() => {
        const {container} = render( <CounterApp value={initialValue} /> );
        expect(container).toMatchSnapshot();
    });

    test('Prueba mostrar el valor inicial de 100',() => {
        render( <CounterApp value={100} /> );
        //expect(screen.getByText(100)).toBeTruthy();
        expect(screen.getByRole('heading',{level:2}).innerHTML).toContain('100');
    });

    test('Debe de incrementar con el botón +1', () => {

        render(<CounterApp value={initialValue} />);
        fireEvent.click( screen.getByText('+1') );
        expect(screen.getByText('2')).toBeTruthy();
    });

    test('Debe de decrementar con el botón -1', () => {

        render(<CounterApp value={initialValue} />);
        fireEvent.click( screen.getByText('-1') );
        //screen.debug();
        expect(screen.getByText('0')).toBeTruthy();
    });

    test('Debe funcionar el botón de reset', () => {
        render(<CounterApp value={initialValue} />);
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        //fireEvent.click( screen.getByText('Reset') );
        fireEvent.click(screen.getByRole('button',{name: 'btn-reset'}));

        expect(screen.getByText(initialValue)).toBeTruthy();
    })


 })