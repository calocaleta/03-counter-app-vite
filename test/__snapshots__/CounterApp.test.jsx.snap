// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el componente CounterApp Prueba hacer match con snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2
    data-testid="test-value"
  >
     
    1
     
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
